### ISSUE TITLE
Maximum call stack size exceeded on JavaScript file format

### ISSUE THREAD

**Issue by edaloicaro18** [prettier/prettier-vscode#138]
- VSCode Version: 1.13.1 (379d2efb5539b09112c793d3d9a413017d736f89)

- OS : Windows 10 Pro

Steps to Reproduce:

1. Open JavaScript file with wrong indentation
2. Hit Shift+Alt+F
3. Maximum call stack size exceeded

I tried disabling each and every extension in VSCode and then enabling them one by one and the only one that gave me that error is this extension.

**Comment by CiGit** [comment]
Do you have a deeply nested code?
Can you share the code which produce this error?

**Comment by edaloicaro18** [comment]
Well I don't have a git repo, but I've got this [sandbox](https://codesandbox.io/embed/zmY3z39p7)

**Comment by CiGit** [comment]
Which file?
I've no trouble with the ones I've tried.
I would prefer If you could copy paste the code in this issue.
Can you share your settings ?

**Comment by edaloicaro18** [comment]
At first, I thought it had something to do with the JSX in my JS files but then tried to format other types of documents containing pure JS or even HTML and the problem was the same.

You can try with all the files inside  the "components" folder

You mean the user settings, right?

```
{
    "workbench.colorTheme": "Cobalt2",
    "workbench.iconTheme": "vs-seti",
    "window.zoomLevel": 0,
    "emmet.syntaxProfiles": {
        "javascript": "jsx"
    },
    "editor.minimap.enabled": false
}
```

**Comment by CiGit** [comment]
Yes those settings. Nothing relevant to this extension.
I've no errors at all.
Can you try to uninstall / reinstall the extension (at that point I don't know really)
Can you also try with `eslintIntegration true`.

**Comment by CiGit** [comment]
Stalled, closing.

**Comment by anantoghosh** [comment]
I also started facing this issue almost randomly. Prettier was working before.
Installing prettier locally fixed it for me.

**Comment by CiGit** [comment]
Which version did you install locally?
Which file did this error?
What are your settings?

**Comment by CiGit** [comment]
I think we have 1 recursive call (find package.json) So if your path is 10k folders deep, it may happen Oo
Maybe in prettier if you have deeply nested calls. dunno

Hypothesis: 10k is still the call stack size :-D

**Comment by anantoghosh** [comment]
I installed prettier 1.7.3 locally.
Threw error on every file.
After about 3 seconds after calling format document.
Default settings.

I'm currently busy with work and can't throughly examine it right now.
It doesn't seem like many people face this problem. I'll try to help once I'm free 🤞

**Comment by anantoghosh** [comment]
```
Maximum call stack size exceeded: RangeError: Maximum call stack size exceeded
    at new fs.Stats (fs.js:133:20)
    at fs.statSync (fs.js:912:18)
    at Object.fs.statSync (ELECTRON_ASAR.js:280:16)
    at typeSync (C:\Users\username\.vscode\extensions\esbenp.prettier-vscode-0.23.1\node_modules\path-type\index.js:18:15)
    at Function.module.exports.sync (C:\Users\username\.vscode\extensions\esbenp.prettier-vscode-0.23.1\node_modules\read-pkg\index.js:38:16)
    at Function.module.exports.sync.opts [as sync] (C:\Users\username\.vscode\extensions\esbenp.prettier-vscode-0.23.1\node_modules\read-pkg-up\index.js:23:16)
    at findPkg (C:\Users\username\.vscode\extensions\esbenp.prettier-vscode-0.23.1\out\src\requirePkg.js:7:27)
    at findPkg (C:\Users\username\.vscode\extensions\esbenp.prettier-vscode-0.23.1\out\src\requirePkg.js:14:16)
    at findPkg (C:\Users\username\.vscode\extensions\esbenp.prettier-vscode-0.23.1\out\src\requirePkg.js:14:16)
    at findPkg (C:\Users\username\.vscode\extensions\esbenp.prettier-vscode-0.23.1\out\src\requirePkg.js:14:16)
```

Currently I have a project with prettier installed locally where there is no problem.
And for rest I get the error above.

**Comment by anantoghosh** [comment]
@CiGit I've found my problem:
findPkg is indeed the culprit.

(on Windows)
My project folder was `D:\Develop\http\`
and I had a package.json file in `D:` which I had just copied from a project.

1st call
![1](https://user-images.githubusercontent.com/12339068/31542323-28e56f96-b02f-11e7-965b-9ad0290d94d3.PNG)

2nd call
![2](https://user-images.githubusercontent.com/12339068/31542324-29330468-b02f-11e7-9770-b110c182b6a3.PNG)

3rd call (this keeps repeating)
![3](https://user-images.githubusercontent.com/12339068/31542325-29c89dc0-b02f-11e7-840d-74a3c5f9fd85.PNG)

`read-pkg-up` with fspath `D:\` keeps finding `D:\package.json`. and
```js
path.resolve(path.dirname(res.path), '..')
```
resolves to `D:\` again.

*Removing the package.json from the root drive (D:) makes prettier-vscode work again*
